tests/plugins/test_topic_checking.py

```diff
@@ -0,0 +1,528 @@
+import pytest
+import logging
+
+from amqtt.plugins.manager import BaseContext
+from amqtt.plugins.topic_checking import (
+    BaseTopicPlugin,
+    TopicTabooPlugin,
+    TopicAccessControlListPlugin,
+    Action,
+)
+from amqtt.session import Session
+
+
+# Base plug-in object
+
+
+@pytest.mark.asyncio
+async def test_base_no_config(logdog):
+    """
+    Check BaseTopicPlugin returns false if no topic-check is present.
+    """
+    with logdog() as pile:
+        context = BaseContext()
+        context.logger = logging.getLogger("testlog")
+        context.config = {}
+
+        plugin = BaseTopicPlugin(context)
+        authorised = plugin.topic_filtering()
+        assert authorised is False
+
+        # Should have printed a couple of warnings
+        log_records = list(pile.drain(name="testlog"))
```

Comment:
The example in the documentation interacts with `pile` after closing the context manager. Not sure how exactly it is intended to be used.